src/Tizen.Core/Tizen.Core/TizenCore.cs

```diff
@@ -0,0 +1,171 @@
+﻿/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+
+namespace Tizen.Core
+{
+    /// <summary>
+    /// The class which provides functions related to the Tizen Core.
+    /// </summary>
+    public static class TizenCore
+    {
+        /// <summary>
+        /// Initializes Tizen Core.
+        /// This method should be called once per application before creating any Tasks.
+        /// Calling this method more than one time will increase internal reference counts which need to be matched by calling Shutdown().
+        /// Failing to call Shutdown() in corresponding number of calls may cause resource leakages.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// 
+        /// TizenCore.Initialize();
+        /// var task = TizenCore.Spawn("Worker");
+        /// 
+        /// </code>
+        /// </example>
+        /// <since_tizen> 12 </since_tizen>
+        static public void Initialize()
+        {
+            Interop.LibTizenCore.TizenCore.Init();
+        }
+
+        /// <summary>
+        /// Shuts down Tizen Core.
+        /// </summary>
+        /// <example>
+        /// <code>
+        /// 
+        /// TizenCore.Shutdown();
+        /// 
+        /// </code>
+        /// </example>
+        /// <since_tizen> 12 </since_tizen>
+        static public void Shutdown()
+        {
+            Interop.LibTizenCore.TizenCore.Shutdown();
+        }
+
+        /// <summary>
+        /// Finds the task instance.
+        /// </summary>
+        /// <param name="id">The ID of the task.</param>
+        /// <example>
+        /// <code>
+        /// 
+        /// TizenCore.Initialize();
+        /// var task = TizenCore.Find("Test") ?? TizenCore.Spawn("Test");
+        /// 
+        /// </code>
+        /// </example>
+        /// <returns>On success the task instance, othwerwise null.</returns>
+        public static Task Find(string id)
+        {
+            return Task.Find(id);
+        }
+
+        /// <summary>
+        /// Finds the task instance from current thread.
+        /// </summary>
+        /// <remarks>
+        /// If the current thread does not have the task, the method tries to find the 'main' task.
```

Comment:
oh, but that would not be the current thread, it's almost contradicting because the returned that might not be running the loop in this thread. For me this is so important difference that I would suggest moving that line from remarks to summary (possibly shortened to "or the "main" task."

Reply:
Done.